news/8850.bugfix.rst

```diff
@@ -0,0 +1,3 @@
+(Linux) Fix discovery and collection of Python shared library when using
+``uv``-installed or ``rye``-installed Python that happens to be of same
+version as the system-installed Python.
```

Comment:
Maybe add something to say that these are still recommended against?

Reply:
I guess if we are going to recommend against specific python builds, the README is probably a better place. 

But I don't see much point, to be honest - we've kept recommending against Anaconda for as long as I'm around here, and it didn't seem to make much of a difference...

Reply:
Fair enough